src/AI/JakeMan.java

```diff
@@ -0,0 +1,770 @@
+  private static final double PEACEFUL_SELF_THREAT_RATIO = 1;
+  private static final int    UNIT_HEAL_THRESHHOLD = 6; // HP at which units heal
+  private static final double UNIT_REFUEL_THRESHHOLD = 1.3; // Factor of cost to get to fuel to start worrying about fuel
+  private static final double UNIT_REARM_THRESHHOLD = 0.25; // Fraction of ammo in any weapon below which to consider resupply
+
+  private Map<UnitModel, Map<XYCoord, Double>> unitMapEnemy;
+  private Map<UnitModel, Map<XYCoord, Double>> unitMapFriendly;
+
+  UnitModel infantry;
+  ArrayList<UnitModel> allTanks;
+  UnitModel antiAir;
+  UnitModel copter;
+
+  public JakeMan(Army army)
+  {
+    super(army);
+
+    // look where all vehicles are and what their threat ranges are (yes, mechs are vehicles)
+    // take free dudes that you have more defenders for than them
+    // move your leftover vehicles so they cover the tiles attacking most relevant stuff (contested cities, own units)
+    // move infs as far towards nearest contested stuff as you can without underdefending, cap if possible
+
+    aiPhases = new ArrayList<AIModule>(
+        Arrays.asList(
+            new PowerActivator(army, CommanderAbility.PHASE_TURN_START),
+            new CapChainActuator(army, this),
+            new CaptureFinisher(army, this),
+            new GenerateThreatMap(army, this), // FreeRealEstate and Travel need this, and NHitKO/building do too because of eviction
+
+            new PowerActivator(army, CommanderAbility.PHASE_BUY),
+            new GetFreeDudes(army, this, false, false), // prioritize non-eviction
+            new GetFreeDudes(army, this, true,  false), // evict if necessary
+            new BuildStuff(army, this),
+            new GetFreeDudes(army, this, true,  true), // step on industries we're not using
+            new Travel(army, this),
+
+            new PowerActivator(army, CommanderAbility.PHASE_TURN_END)
+            ));
+  }
+
+  @Override
+  public void initTurn(GameMap gameMap)
+  {
+    super.initTurn(gameMap);
+    if( null == capPhase )
+      init(gameMap);
+    log(String.format("[======== JakeMan initializing turn %s for %s =========]", turnNum, myArmy));
+  }
+
+  @Override
+  public void endTurn()
+  {
+    super.endTurn();
+    log(String.format("[======== JakeMan ending turn %s for %s =========]", turnNum, myArmy));
+  }
+
+  public static class GenerateThreatMap implements AIModule
+  {
+    private static final long serialVersionUID = 1L;
+    public Army myArmy;
+    public final JakeMan ai;
+
+    public GenerateThreatMap(Army co, JakeMan ai)
+    {
+      myArmy = co;
+      this.ai = ai;
+    }
+
+    @Override
+    public GameAction getNextAction(PriorityQueue<Unit> unitQueue, GameMap gameMap)
+    {
+      ai.unitMapEnemy = new HashMap<UnitModel, Map<XYCoord, Double>>();
+      ai.unitMapFriendly = new HashMap<UnitModel, Map<XYCoord, Double>>();
```

Comment:
This completely rebuilds the threat map for every command the AI returns? Seems... inefficient.